manager/allocator/network.go

```diff
@@ -641,6 +644,35 @@ func (a *Allocator) commitAllocatedNode(ctx context.Context, batch *store.Batch,
 	return nil
 }
 
+func updatePortsInHostPublishMode(s *api.Service) {
```

Comment:
@yongtang Can you please add a comment on what this function does. I think it would be beneficial for understanding the code flow, in future.

I am thinking of something like 
```
This function prepares the service object for being updated when the change regards 
the published ports in host mode: It resets the runtime state ports (s.Endpoint.Ports)
to the current ingress mode runtime state ports plus the newly configured publish mode ports,
so that the service allocation invoked on this new service object will trigger the deallocation
of any old publish mode port and allocation of any new one.
```

or whatever you think is appropriate to explain to the reader the intention of this change.